test/Makefile

```diff
@@ -1,8 +1,39 @@
-ARCH ?= x86_64
-LIB_PATH ?= dist/linux/release/$(ARCH)/libnorddrop.so
 
+FEATURES ?= --features moose_file
+LIB ?= ../target/debug/libnorddrop.so
+LIB_DOCKER=${LIB}
+COVERAGE_PATH ?= coverage
+HTML_PATH ?= html
+RUSTFLAGS_COVERAGE ?= -C instrument-coverage -Zprofile -Copt-level=0 -Clink-dead-code -Coverflow-checks=off
+
+.DEFAULT_GOAL := test
+.PHONY: build build-coverage test coverage clean
+
+# Build the project
 build:
-	../../ci/build.py build linux $(ARCH)
+	cargo build $(FEATURES)
+
+# Build the project with coverage instrumentation
+build-coverage:
+	CARGO_INCREMENTAL=0 RUSTFLAGS="$(RUSTFLAGS_COVERAGE)" cargo build $(FEATURES)
+
+# Test the project
+test: build
+	LIB_PATH=$(LIB_DOCKER) ./runner.py
+
+# Generate and view coverage report
+coverage: build-coverage
+	mkdir -p ${COVERAGE_PATH}
+
+	# allow coverage of failed test cases
+	LIB_PATH=${LIB_DOCKER} ./runner.py || true
+
+	llvm-profdata merge -sparse $(COVERAGE_PATH)/*.profraw -o full.profdata
+	llvm-cov report -instr-profile=full.profdata --ignore-filename-regex='/.cargo/registry' --Xdemangler=rustfilt $(LIB)
+	grcov $(COVERAGE_PATH) --binary-path $(LIB) -s .. -t ${HTML_PATH}
+	python -m http.server 8080 --directory ./html
```

Comment:
Isn't it better to just open the HTML file in the browser?

Reply:
Not sure if `open` is present everywhere..